### PR TITLE
chore(tests): setup circleci to run unit tests againt v5

### PR DESCRIPTION
**Pull request by pieh**

### CODE REVIEW

File: packages/gatsby/src/query/__tests__/file-parser.js [hunk 1]
```diff
@@ -303,10 +306,11 @@ export const config = async () => {
...
 
-    // The second param is a "hint", see: https://jestjs.io/docs/en/expect#tomatchsnapshotpropertymatchers-hint
-    expect(results).toMatchSnapshot({}, `results`)
-    expect(reporter.warn.mock.calls).toMatchSnapshot({}, `warn`)
-    expect(reporter.panicOnBuild.mock.calls).toMatchSnapshot({}, `panicOnBuild`)
+    expect({
+      results,
+      warnCalls: reporter.warn.mock.calls,
+      panicOnBuildCalls: reporter.panicOnBuild.mock.calls,
+    }).toMatchSnapshot()
```
Comment (pieh) [inline packages/gatsby/src/query/__tests__/file-parser.js:313]: this was failing with "obsolete snapshots" even tho jest is supposed to not fail on it when there are snapshots in skipped tests ( https://github.com/facebook/jest/issues/3038 ). Seems like either using multiple snapshots or using the named ones is edge case, so I used single snapshot here instead

File: packages/gatsby/src/schema/types/sort.ts [hunk 2]
```diff
@@ -193,7 +193,7 @@ export const getSortInputNestedObjects = ({
   const itc = convertToNestedInputType({
     schemaComposer,
     typeComposer,
-    postfix: `SortInputProp2`,
+    postfix: `SortInput`,
```
Comment (pieh) [inline packages/gatsby/src/schema/types/sort.ts:196]: This is actual code fix that slipped through original PR ( https://github.com/gatsbyjs/gatsby/pull/36708 )

**Review by j0sh77 — APPROVED** [review]
:shipit:
